File: backend/app_api/domain/entities/Bitacora_entity.py
from dataclasses import dataclass
from app_api.domain.values.timestamp import TimestampModel

@dataclass(kw_only=True)
class BitacoraEntity(TimestampModel):
    id: int | None
    usuario: int
    accion: str
    ipv4: str
    nivel: str

File: frontend/src/domain/entities/Bitacora_entity.ts
import { TimeStamp } from "../values/Timestamp";

export class BitacoraEntity extends TimeStamp{
    id: number | null;
    usuario: number;
    accion: string;
    ipv4: string;
    nivel: string;

    constructor(
        id: number | null, 
        usuario: number, 
        accion: string, 
        ipv4: string, 
        nivel: string,
        created_at: Date | null = null,
        updated_at: Date | null = null,
        deleted_at: Date | null = null,
    ) {
        super(created_at, updated_at, deleted_at);
        this.id = id;
        this.usuario = usuario;
        this.accion = accion;
        this.ipv4 = ipv4;
        this.nivel = nivel;
    }
}

